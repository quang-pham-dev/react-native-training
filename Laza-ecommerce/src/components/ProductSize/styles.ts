import { StyleSheet } from 'react-native';

// Themes
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';
import Colors from 'themes/Colors';

const styles = StyleSheet.create({
  sizeContainer: {
    justifyContent: 'space-between',
  },

  textSizeWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },

  textSize: {
    fontSize: Fonts.size.default,
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    lineHeight: Fonts.lineHeight.sm,
    color: Colors.textBlack,
  },

  textSizeGuide: {
    fontSize: Fonts.size.default,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    lineHeight: Fonts.lineHeight.base,
    color: Colors.primaryColor,
  },

  sizeWrapper: {
    flexDirection: 'row',
  },

  sizeItem: {
    width: 60,
    height: 60,
    borderRadius: Metrics.borderRadius.roundedMedium,
    backgroundColor: Colors.secondaryBackground,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 8,
    marginTop: 10,
  },
});

export default styles;
