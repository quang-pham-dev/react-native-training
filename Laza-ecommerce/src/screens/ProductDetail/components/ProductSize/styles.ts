import { StyleSheet } from 'react-native';

// Themes
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';
import Colors from 'themes/Colors';

const styles = StyleSheet.create({
  sizeContainer: {
    width: '100%'
  },

  textSizeWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },

  textSizeGuide: {
    fontSize: Fonts.size.default,
    color: Colors.primaryColor
  },

  sizeWrapper: {
    width: '100%',
    marginTop: 10
  },

  sizeItem: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 60,
    height: 60,
    borderRadius: Metrics.borderRadius.roundedMedium,
    backgroundColor: Colors.secondaryBackground,
    marginRight: 10
  }
});

export default styles;
