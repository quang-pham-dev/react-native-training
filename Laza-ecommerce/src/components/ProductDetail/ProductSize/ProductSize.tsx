import React from 'react';
import { PRODUCTS_SIZE } from 'constants/Products';
import { View, ScrollView, Pressable, Text } from 'react-native';
import styles from './styles';

export function ProductSize() {
  return (
    <View style={styles.sizeContainer}>
      <View style={styles.textSizeWrapper}>
        <Text style={styles.textSize}>Size</Text>
        <Text style={styles.textSizeGuide}>Size Guide</Text>
      </View>
      <ScrollView style={styles.sizeWrapper} horizontal showsHorizontalScrollIndicator={false}>
        {PRODUCTS_SIZE.map((item: string) => (
          <Pressable style={styles.sizeItem} key={item}>
            <Text style={styles.textSize}>{item}</Text>
          </Pressable>
        ))}
      </ScrollView>
    </View>
  );
}
