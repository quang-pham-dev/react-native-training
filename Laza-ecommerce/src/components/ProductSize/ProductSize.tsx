import React from 'react';
import { View, ScrollView, Pressable, Text } from 'react-native';

// Constants
import { PRODUCTS_SIZE } from 'constants/Products';

// Styles
import styles from './styles';

const ProductSize = () => {
  return (
    <View style={styles.sizeContainer}>
      <View style={styles.textSizeWrapper}>
        <Text style={styles.textSize}>Size</Text>
        <Text style={styles.textSizeGuide}>Size Guide</Text>
      </View>
      <ScrollView style={styles.sizeWrapper} horizontal showsHorizontalScrollIndicator={false}>
        {PRODUCTS_SIZE.map((item: string) => (
          <Pressable style={styles.sizeItem} key={item}>
            <Text style={styles.textSize}>{item}</Text>
          </Pressable>
        ))}
      </ScrollView>
    </View>
  );
};

export default ProductSize;
