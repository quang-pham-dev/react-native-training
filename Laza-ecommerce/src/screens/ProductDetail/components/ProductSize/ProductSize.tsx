import React, { memo } from 'react';
import { View, ScrollView, Pressable, Text } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Constants
import { IProductDetailItemProps } from 'types/screens/ProductDetail';
import { ProductSizeProps } from 'types/models/Products';

// Styles
import styles from './styles';

const ProductSize = ({ product }: IProductDetailItemProps) => {
  const { size } = product || {};

  return (
    <View style={styles.sizeContainer}>
      <View style={styles.textSizeWrapper}>
        <Text style={styles.textSize}>Size</Text>
        <Text style={styles.textSizeGuide}>Size Guide</Text>
      </View>

      <ScrollView style={styles.sizeWrapper} horizontal showsHorizontalScrollIndicator={false}>
        {Boolean(size) &&
          size.map((item: ProductSizeProps) => (
            <Pressable style={styles.sizeItem} key={item.id}>
              <Text style={styles.textSize}>{item.size}</Text>
            </Pressable>
          ))}
      </ScrollView>
    </View>
  );
};

export default memo(ProductSize, isEqual);
