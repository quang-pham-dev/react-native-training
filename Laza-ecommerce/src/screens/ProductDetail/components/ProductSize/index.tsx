import React, { memo } from 'react';
import { View, ScrollView, Pressable, Text } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Components
import Title from 'components/Title';

// Constants
import { IProductDetailItemProps } from 'types/screens/ProductDetail';

// Types
import { ProductSizeProps } from 'types/models/Products';

// Styles
import styles from './styles';

const ProductSize = ({ sizes }: IProductDetailItemProps) => {
  return (
    <View style={styles.sizeContainer}>
      <View style={styles.textSizeWrapper}>
        <Title.titleLeftSection titleName='Size' />
        <Title.titleRightSection style={styles.textSizeGuide}>Size Guide</Title.titleRightSection>
      </View>

      <ScrollView style={styles.sizeWrapper} horizontal showsHorizontalScrollIndicator={false}>
        {Boolean(sizes) &&
          sizes?.map((item: ProductSizeProps) => (
            <Pressable style={styles.sizeItem} key={item.id}>
              <Title titleName={item.size} />
            </Pressable>
          ))}
      </ScrollView>
    </View>
  );
};

export default memo(ProductSize, isEqual);
