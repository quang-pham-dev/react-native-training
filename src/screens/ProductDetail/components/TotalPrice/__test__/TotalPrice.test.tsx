import React from 'react'

// LIBS
import renderer from 'react-test-renderer'

// Components
import ProductSize from '@screens/ProductDetail/components/Sizes'

// Constants
import {product} from '@constants'

const props = {
  sizes: product.sizes,
}

describe('Product detail TotalPrice', () => {
  const tree = renderer.create(<ProductSize {...props} />)

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot()
  })
})
