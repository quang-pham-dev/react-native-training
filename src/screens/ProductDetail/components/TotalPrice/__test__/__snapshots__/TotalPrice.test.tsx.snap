// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product detail TotalPrice should render correctly 1`] = `
<View
  pVertical={15}
  style={
    Array [
      Object {
        "paddingVertical": 15,
        "width": "100%",
      },
    ]
  }
>
  <View
    pBottom={10}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1D1E20",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 17,
            "lineHeight": 18.7,
            "textAlign": "left",
          },
        ]
      }
      type="productDetailTitleSection"
    >
      Size
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#3D93F8",
            "fontFamily": "Inter-Regular",
            "fontSize": 15,
            "lineHeight": 16.5,
            "textAlign": "left",
          },
        ]
      }
      type="productDetailSizeGuide"
    >
      Size Guide
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        pRight={9}
        style={
          Array [
            Object {
              "paddingRight": 9,
            },
          ]
        }
      >
        <View
          bRadius={10}
          bgColor="#F5F6FA"
          h={60}
          pHorizontal={20}
          pVertical={20}
          style={
            Array [
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "height": 60,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": 60,
              },
            ]
          }
          w={60}
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            S
          </Text>
        </View>
      </View>
      <View
        pRight={9}
        style={
          Array [
            Object {
              "paddingRight": 9,
            },
          ]
        }
      >
        <View
          bRadius={10}
          bgColor="#F5F6FA"
          h={60}
          pHorizontal={20}
          pVertical={20}
          style={
            Array [
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "height": 60,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": 60,
              },
            ]
          }
          w={60}
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            M
          </Text>
        </View>
      </View>
      <View
        pRight={9}
        style={
          Array [
            Object {
              "paddingRight": 9,
            },
          ]
        }
      >
        <View
          bRadius={10}
          bgColor="#F5F6FA"
          h={60}
          pHorizontal={20}
          pVertical={20}
          style={
            Array [
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "height": 60,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": 60,
              },
            ]
          }
          w={60}
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            L
          </Text>
        </View>
      </View>
      <View
        pRight={9}
        style={
          Array [
            Object {
              "paddingRight": 9,
            },
          ]
        }
      >
        <View
          bRadius={10}
          bgColor="#F5F6FA"
          h={60}
          pHorizontal={20}
          pVertical={20}
          style={
            Array [
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "height": 60,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": 60,
              },
            ]
          }
          w={60}
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            XL
          </Text>
        </View>
      </View>
      <View
        pRight={9}
        style={
          Array [
            Object {
              "paddingRight": 9,
            },
          ]
        }
      >
        <View
          bRadius={10}
          bgColor="#F5F6FA"
          h={60}
          pHorizontal={20}
          pVertical={20}
          style={
            Array [
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "height": 60,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": 60,
              },
            ]
          }
          w={60}
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            2XL
          </Text>
        </View>
      </View>
      <View
        pRight={9}
        style={
          Array [
            Object {
              "paddingRight": 9,
            },
          ]
        }
      >
        <View
          bRadius={10}
          bgColor="#F5F6FA"
          h={60}
          pHorizontal={20}
          pVertical={20}
          style={
            Array [
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "height": 60,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": 60,
              },
            ]
          }
          w={60}
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            3XL
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
